### PR TITLE
refactor: push chunk and partition summary into catalog

### PR DESCRIPTION
**Pull request by tustvold**
Pushes some of the aggregation logic from Db into catalog so that it can be used by the system tables more easily.

### CODE REVIEW

File: server/src/db/catalog.rs [hunk 1]
```diff
@@ -134,6 +135,15 @@ impl Catalog {
         chunks
     }
 
+    pub fn chunk_summaries(&self) -> Vec<ChunkSummary> {
+        let mut summaries = Vec::new();
```
Comment (tustvold) [inline server/src/db/catalog.rs:139]: Can't use flat_map as partition.chunk_summaries returns a borrowing iterator

Comment (alamb) [reply server/src/db/catalog.rs:139]: I think you would have to use `collect()` at the end to pull everything into a `Vec`

I don't think it matters practically one way or the other

**Review by alamb — APPROVED** [review]
Looks good to me

File: server/src/db/catalog.rs (hunk 1 above)
Comment (tustvold) [reply server/src/db/catalog.rs:139]: Yeah, you have to have an intermediate buffer. I'd take avoiding that over functional purity :grin:

**Pull request merged**